src/documentation/programming-guide.md

```diff
@@ -917,7 +917,94 @@ While `ParDo` always produces a main output `PCollection` (as the return value f
 
 ## <a name="transforms-composite"></a>Composite Transforms
 
-> **Note:** This section is in progress ([BEAM-1452](https://issues.apache.org/jira/browse/BEAM-1452)).
+Transforms can have a nested structure, where a complex transform performs multiple simpler transforms (such as more than one `ParDo`, `Combine`, or `GroupByKey`). These transforms are called composite transforms. Nesting multiple transforms inside a single composite transform can make your code more modular and easier to understand.
```

Comment:
This sort of implies one level of nesting. You might want to end the comma-separated list with something like "or even other composite transforms" to make the point that abstractions can be layered here.